zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/writers/TypedResponseWriter.java

```diff
@@ -24,6 +25,16 @@ void writeRejection(
       final long requestId,
       final int requestStreamId);
 
+  void writeRejection(
+      final Intent intent,
+      final long key,
```

Comment:
🔧 Let's have a `key` as the first parameter, this will align with `writeEventOnCommand` & `writeResponse` methods of this interface.